docs/docs/custom-fields.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Custom fields
+tags:
+  - content-type-builder
+  - plugins
+---
+
+# Custom fields
+
+## Summary
+
+Custom fields provide a way to replace the inputs of existing Strapi types to improve the content editing experience.
+
+## Detailed design
+
+A custom field needs to be registered in both the admin and server.
+
+### Server
+
+We need to ensure that an attribute using a custom field is valid, we expose a new `customFields` object with a `register` method on the `Strapi` instance.
+
+Here's how you register a Custom Field on the server:
```

Comment:
I think these can be deleted when we move the block up here that links to the documentation